webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx

```diff
@@ -63,17 +83,20 @@ export const TableColumnHeader: React.FC<HeaderRendererProps<any>> = observer(fu
 
   // TODO we want to get "sortable" property from SqlResultColumn data
   const sortable = model.source.results.length === 1;
-  const columnTooltip = columnName + (column?.fullTypeName ? ': ' + column.fullTypeName : '');
+  const readOnly = !tableDataContext.isReadOnly() && column?.readOnly;
+  const readOnlyStatus = readOnly && column?.readOnlyStatus ? `(Read-only: ${column.readOnlyStatus})` : '';
+  const columnTooltip = `${columnName}${column?.fullTypeName ? ': ' + column.fullTypeName : ''} ${readOnlyStatus}`;
```

Comment:
show readOnlyStatus if it is presented, the only icon should be shown if the table not in read-only mode